lib/plugins/aws/remove/lib/bucket.js

```diff
@@ -75,21 +71,23 @@ module.exports = {
   async deleteObjects() {
     legacy.log('Removing objects in S3 bucket...');
     if (this.objectsInBucket.length) {
-      return this.provider.request('S3', 'deleteObjects', {
+      await this.provider.request('S3', 'deleteObjects', {
         Bucket: this.bucketName,
         Delete: {
           Objects: this.objectsInBucket,
         },
       });
     }
-
-    return BbPromise.resolve();
   },
 
   async emptyS3Bucket() {
-    return BbPromise.bind(this)
-      .then(this.setServerlessDeploymentBucketName)
-      .then(this.listObjects)
-      .then(this.deleteObjects);
+    await this.setServerlessDeploymentBucketName();
+    if (this.bucketName && (await this.checkIfBucketExists(this.bucketName))) {
```

Comment:
Wouldn't `bucketName` be unconditionally set here? (we didn't have that check before)

Reply:
There is a scenario in which it won't be set so we need to leave this check this way

Reply:
Is this a scenario that was already in `master` and was not handled? Or is this scenario introduced with this PR?

Also when exactly is that the case?

Reply:
Yes, in `master` it was just throwing an error on SDK call in `setServerlessDeploymentBucket`, now we want to handle such situation gracefully